lib/nerves_hub/console_channel.ex

```diff
@@ -174,12 +174,25 @@ defmodule NervesHub.ConsoleChannel do
     state
   end
 
-  # All other requests push IO Request over the socket. 
+  # All other requests push IO Request over the socket.
   defp io_request(from, reply_as, {:get_line, encoding, data} = req, state) do
     Channel.push_async(state.channel, "get_line", %{encoding: encoding, data: data})
     %{state | request: {from, reply_as, req}}
   end
 
+  defp io_request(from, reply_as, {:put_chars, encoding, module, function, args}, state) do
+    # Judicious try block so we can return back the error tuple
+    # as defined in http://erlang.org/doc/apps/stdlib/io_protocol.html#output-requests
+    try do
+      data = apply(module, function, args)
+      io_request(from, reply_as, {:put_chars, encoding, data}, state)
+    catch
+      kind, value ->
+        Client.handle_error(@client, {kind, value})
+        io_reply(from, reply_as, {:error, {kind, value}})
```

Comment:
This catches more than what the doc says that you should - i.e., it catches exits and errors. I read it as they wanted you to catch thrown values so the code would look like this:

```
try do
...
catch
  thrown_value ->
    Client.handle_error(@client, {:error, thrown_value})
    io_reply(from, reply_as, {:error, thrown_value})
end
```

Reply:
Updated